### DISCUSSION TITLE
Custom new tab page URL

### DISCUSSION THREAD

**Discussion by MatiasDesuu** [zen-browser/desktop#1647] [Ideas]
This is basically the only feature that keeps me away from Firefox-based browsers.
Yes, you can set a Home Page default URL, but the new page is always blank or browser defaults home page.

Would love the option to set a new tab page URL, that way we can set, for example, a self-hosted dashboard as our new tab page :)

**Comment by takenotice** [comment]
I've just discovered this. Much bummer. I've just started to love the Tab Stash Add-On and wanted this for new tabs, but i guess not.

**Comment by blakegearin** [comment]
Isn't that the functionality the [New Tab Override extension](https://addons.mozilla.org/en-US/firefox/addon/new-tab-override/) provides?

**Reply by nekoliten** [reply]
yes and now, it does allow for custom url's in new tabs, but it doesn't allow for automatically clearing the address bar if you start typing something. The creator of the extension even highlights it in their notes, saying it is not possible to implement without a change to the browser by Mozilla.

**Comment by noelpuig** [comment]
You can do this with extensions, as mentioned, they support adding custom URLs, but they do not allow access to local files, which means you cannot load your own new-page HTML directly.

I managed to work around it. The HTML that the new tab renders is located in:

`C:\Program Files\Zen Browser\browser\omni.ja\chrome\browser\builtin-addons\newtab\prerendered\activity-stream-noscripts.html` (For windows of course)

The `omni.ja` file is actually a ZIP file. To open it, simply rename it back to `.zip`, extract it, replace the HTML files, recompress it, and rename it back to `.ja`. The browser does not allow importing external files, so any scripts, styles, and images must be embedded directly in the HTML (In my case, images in base64). Even though it's named `-noscripts`, it does support embedded js.

<img width="2256" height="1423" alt="image" src="https://github.com/user-attachments/assets/10fe4e23-5fb8-4cf6-8bdb-2c629f472851" />
